core/src/entities/packages/package.rs

```diff
@@ -14,20 +17,41 @@ pub struct Package {
     /// The unique identifier for the Package.
     pub id: String,
 
+    /// Discriminator indicating if the Package defines a [Package] being monitored or a dependency
+    /// of a [Package] being monitored.
+    pub kind: PackageKind,
```

Comment:
Awesome.  That's what I think this should be too.  I love it when a plan comes together